src/main/scala/dpla/ingestion3/enrichments/Spatial.scala

```diff
@@ -0,0 +1,172 @@
+
+case class Bounds(ne: Coords, sw: Coords)
+
+case class Geometry(center: Coords, bounds: Bounds)
+
+case class Feature(geometry: Geometry, woeType: Int, displayName: String)
+
+case class Interpretation(feature: Feature)
+
+case class Result(interpretations: List[Interpretation])
+
+object Spatial extends MappingUtils with DefaultVocabularies {
+
+  implicit val formats = DefaultFormats // Formats for json4s
+  val log = Logger.getLogger(getClass.getName)
+
+  def enrich(m: Model): Model = {
+
+    val spatialModel = m.filter(null, rdf.`type`, edm.Place)
+    if (spatialModel.isEmpty) return m
+
+    for {
+      s <- spatialModel.subjects
+      provLabels = m.filter(s, dpla.providedLabel, null)
+      pl <- provLabels
+    } {
+
+      /* We expect the provided label below not to need any special processing,
+       * like stripping newlines, etc. That should be handled by an earlier
+       * enrichment.
+       *
+       * If it's a multiline string, as we've encountered with Digital
+       * Commonwealth, I really don't know if we can handle it well. Twofishes
+       * will not do what you want with a string like "Dallas Dallas Texas
+       * United States North and Central America". This particular instance will
+       * work if the values are reversed (as "North and Central America United
+       * States Texas Dallas Dallas") but I wouldn't want to count on that in
+       * general.
+       *
+       * We need some research to determine how common this is. Ideally we'd
+       * have multiple dc:spatial statements instead of one statement with a
+       * multiline string as the object.
+       *
+       * Is this an issue with Krikri that won't happen with Ingestion 3?
+       */
+
+      try {
+
+        val providedLabel = pl.getObject.stringValue
+        m.add(s, dpla.providedLabel, literal(providedLabel))
+
+        // Look up the place by name (providedLabel) ...
+        val baseURI = "http://geo-prod:8081/query"
+        val response: HttpResponse[String] =
+          Http(baseURI).param("lang", "en")
+            .param("responseIncludes", "PARENTS,DISPLAY_NAME")
+            .param("query",  providedLabel).asString
+        val jsonDoc = parse(response.body)
+
+        val name = ((jsonDoc \ "interpretations") (0) \ "feature")
+          .extract[Feature].displayName
+        val coords = ((((jsonDoc \ "interpretations") (0) \ "feature") \ "geometry") \ "center")
+          .extract[Coords]
+        m.add(s, skos.prefLabel, literal(name))
+        m.add(s, wgs84.lat, literal(coords.lat.toString))
+        m.add(s, wgs84.long, literal(coords.lng.toString))
+
+        // Next do something with parents (requires adding case classes above
+        // for parent features, etc.)
+
+      } catch {
+        // Best way to log a message?
+        case e: java.lang.IndexOutOfBoundsException => {
+          log.error(s"No interpretations for ${pl.getObject.stringValue}")
+        }
+        case e: Throwable => {
```

Comment:
I'll change that.

I was responding to a warning from the compiler about how I should change something that I'd written at some point (can't remember exactly what it was) to define `e` as `Throwable`, I think because I may have had it untyped.